dev-utils/src/commands/genesis_command.rs

```diff
@@ -23,79 +29,240 @@ pub enum GenesisCommand {
         about = "Generate new genesis.json parameters (app_hash + app_state)"
     )]
     Generate {
+        #[structopt(
+            name = "tendermint_genesis_path",
+            short,
+            long,
+            help = "Path to the Tendermint genesis.json file (e.g. ~/.tendermint/config/genesis.json)"
+        )]
+        tendermint_genesis_path: PathBuf,
+
         #[structopt(
             name = "genesis_dev_config_path",
             short,
             long,
             help = "Path to a file containing the genesis-related configuration (e.g. ERC20 holdership) -- see example-dev-conf.json"
         )]
         genesis_dev_config_path: PathBuf,
+
+        #[structopt(
+            name = "inline",
+            short,
+            long,
+            help = "Replace Tenderint genesis.json file in place, saving backups with .bak.json extension"
+        )]
+        inline: bool,
     },
 }
 
 impl GenesisCommand {
     pub fn execute(&self) -> Result<()> {
         match self {
             GenesisCommand::Generate {
+                tendermint_genesis_path,
                 genesis_dev_config_path,
-            } => GenesisCommand::generate(&genesis_dev_config_path).map(|_| ()),
+                inline,
+            } => generate_genesis_command(
+                &tendermint_genesis_path,
+                &genesis_dev_config_path,
+                *inline,
+            )
+            .map(|_| ()),
         }
     }
+}
 
-    pub fn do_generate(genesis_dev: &GenesisDevConfig) -> Result<(String, InitConfig)> {
-        let mut dist: BTreeMap<RedeemAddress, (StakedStateDestination, Coin)> = BTreeMap::new();
+#[derive(Debug, Clone, Serialize)]
+pub struct Validator {
```

Comment:
Instead of creating this `struct` manually. You can use the one in `tendermint-rs`. Currently, its not re-exported from `client-common` but you can do so by adding following line in `client-common/src/tendermint/types.rs`:

```
pub use tendermint::validator::Info as ValidatorInfo;
```

After that, you can do `use client_common::tendermint::types::ValidatorInfo` in `dev-utils`. You may also need to add few more functions via extension traits.

Reply:
Great, I am reexporting those structs now and add necessary traits.

Reply:
After I tried, I decided to add `TendermintValidator` into `chain-core/src/state/tendermint.rs` instead of re-exporting types from `tendermint-rs`.

The reason is that the command now uses structs implemented in `chain-core/src/state/tendermint.rs`. If I introduce the `tendermint-rs` types, it will be using a hybrid of custom-defined Tendermint types and from `tendermint-rs`.

I will left this as a TODO before we migrate everything to use `tendermint-rs` types.